modules/renter/persist.go

```diff
@@ -195,32 +202,37 @@ func (r *Renter) save() error {
 
 // load fetches the saved renter data from disk.
 func (r *Renter) load() error {
-	// Load all files found in renter directory.
-	dir, err := os.Open(r.persistDir) // TODO: store in a subdir?
-	if err != nil {
-		return err
-	}
-	defer dir.Close()
-	filenames, err := dir.Readdirnames(-1)
-	if err != nil {
-		return err
-	}
-	for _, path := range filenames {
-		// Skip non-sia files.
-		if filepath.Ext(path) != ShareExtension {
-			continue
-		}
-		file, err := os.Open(filepath.Join(r.persistDir, path))
+	// Recursively load all files found in renter directory. Errors
+	// encountered during loading are logged, but are not considered fatal.
+	err := filepath.Walk(r.persistDir, func(path string, info os.FileInfo, err error) error {
+		// These errors should be considered fatal.
 		if err != nil {
-			// maybe just skip?
 			return err
 		}
+
+		// Skip folders and non-sia files.
+		if info.IsDir() || filepath.Ext(path) != ShareExtension {
```

Comment:
empty folders are even trickier than empty files. At the moment we don't support creating folders at all (only creating files within folders), so I think it's fine for now.
Later I suppose we'll add support for folder operations, and decide what makes sense then. As usual, I plan to find out how Dropbox does it.